File: Library/src/main/java/org/endeavourhealth/library/model/imq/Expression.java
package org.endeavourhealth.library.model.imq;

public class Expression {

  private String value;
  private String nodeRef;
  private String iri;
  private String propertyRef;

  public String getValue() {
    return value;
  }

  public Expression setValue(String value) {
    this.value = value;
    return this;
  }

  public String getNodeRef() {
    return nodeRef;
  }

  public Expression setNodeRef(String nodeRef) {
    this.nodeRef = nodeRef;
    return this;
  }

  public String getIri() {
    return iri;
  }

  public Expression setIri(String iri) {
    this.iri = iri;
    return this;
  }

  public String getPropertyRef() {
    return propertyRef;
  }

  public Expression setPropertyRef(String propertyRef) {
    this.propertyRef = propertyRef;
    return this;
  }
}

File: src/interfaces/Expression.ts
import z from "zod";

// export interface Expression {
//   value?: string;
//   nodeRef?: string;
//   iri?: string;
//   propertyRef?: string;
// }

export const ExpressionSchema = z.strictObject({
  value: z.string().optional(),
  nodeRef: z.string().optional(),
  iri: z.string().optional(),
  propertyRef: z.string().optional()
});

export type Expression = z.output<typeof ExpressionSchema>;
